core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -56,7 +56,7 @@ object Partitioner {
    */
   def defaultPartitioner(rdd: RDD[_], others: RDD[_]*): Partitioner = {
     val bySize = (Seq(rdd) ++ others).sortBy(_.partitions.size).reverse
-    for (r <- bySize if r.partitioner.isDefined) {
+    for (r <- bySize if r.partitioner.isDefined && r.partitions.nonEmpty) {
       return r.partitioner.get
     }
```

Comment:
And?  Again, if `r.partitioner` is not defined (i.e., is `None`), then the for comprehension I wrote will not execute the `return`.  Only if `r.partitioner` is `Some(p)` && `r.partitions.nonEmpty` will `return p` be executed.  If those conditions are not true for any `r`, then the for comprehension won't do anything and execution will fall through.